src/api/procedures/configureDividendDistribution.ts

```diff
@@ -0,0 +1,239 @@
+    perShare,
+    maxAmount,
+    paymentDate,
+    expiryDate,
+    checkpoint,
+    ...corporateActionArgs
+  } = args;
+
+  if (currency === ticker) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Cannot distribute Dividends using the Security Token as currency',
+    });
+  }
+
+  if (paymentDate <= new Date()) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Payment date must be in the future',
+    });
+  }
+
+  if (expiryDate && expiryDate < paymentDate) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Expiry date must be after payment date',
+    });
+  }
+
+  if (!(checkpoint instanceof Checkpoint)) {
+    let checkpointDate;
+
+    if (checkpoint instanceof Date) {
+      checkpointDate = checkpoint;
+    } else {
+      ({ nextCheckpointDate: checkpointDate } = await checkpoint.details());
+    }
+
+    if (checkpointDate >= paymentDate) {
+      throw new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'Payment date must be after the Checkpoint date',
+      });
+    }
+  }
+
+  const [{ total, locked }] = await portfolio.getTokenBalances({ tokens: [currency] });
+  const freeBalance = total.minus(locked);
+
+  if (freeBalance.lt(maxAmount)) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Origin Portfolio free balance is not enough to cover the distribution amount',
+      data: {
+        freeBalance,
+      },
+    });
+  }
+
+  const caId = await this.addProcedure(initiateCorporateAction, {
+    ticker,
+    kind: CorporateActionKind.UnpredictableBenefit,
+    checkpoint,
+    ...corporateActionArgs,
+  });
+
+  const rawPortfolioNumber = originPortfolio && numberToU64(originPortfolio.id, context);
+  const rawCurrency = stringToTicker(currency, context);
+  const rawPerShare = numberToBalance(perShare, context);
+  const rawAmount = numberToBalance(maxAmount, context);
+  const rawPaymentAt = dateToMoment(paymentDate, context);
+  const rawExpiresAt = expiryDate && dateToMoment(expiryDate, context);
+
+  const [dividendDistribution] = this.addTransaction(
+    tx.capitalDistribution.distribute,
+    {
+      resolvers: [createDividendDistributionResolver(context)],
+    },
+    caId,
+    rawPortfolioNumber,
+    rawCurrency,
+    rawPerShare,
+    rawAmount,
+    rawPaymentAt,
+    rawExpiresAt
+  );
+
+  return dividendDistribution;
+}
+
+/**
+ * @hidden
+ */
+export function getAuthorization(
+  this: Procedure<Params, DividendDistribution, Storage>,
+  { ticker }: Params
+): ProcedureAuthorization {
+  const {
+    storage: { portfolio },
+  } = this;
+
+  return {
+    identityRoles: [
+      { type: RoleType.TokenCaa, ticker },
+      { type: RoleType.PortfolioCustodian, portfolioId: portfolioToPortfolioId(portfolio) },
+    ],
+    signerPermissions: {
+      transactions: [TxTags.capitalDistribution.Distribute],
+      tokens: [],
```

Comment:
I think you need Token permission. Please check.

Reply:
For this you do, but not for creating the CA